lib/restful_resource/http_client.rb

```diff
@@ -270,6 +276,7 @@ def http_request(request)
 
     def handle_error(request, response)
       case response[:status]
+      when 400 then raise HttpClient::BadRequest.new(request, response)
```

Comment:
This will change the behaviour of `400` responses on consumers.

Anything handling `OtherHttpError` but not `HttpError` will now raise loud.

Reply:
Yes, it is super easy to fix and more descriptive.
400 are common errors that we may want to handle different or just add the exception to the rescue 🤷‍♂ 

On the other hand, I understand if we don't want to add noise in this PR with this new error class.

Reply:
Done, I removed the 400 error from this PR and address it in the future